### PR TITLE
spirv-tools: fix build for Linux

### PR DESCRIPTION
**Pull request by danielnachun**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/spirv-tools.rb [hunk 1]
```diff
@@ -50,8 +52,19 @@ def install
 
   test do
     cp libexec/"examples"/"main.cpp", "test.cpp"
+
+    args = []
```
Comment (carlocab) [inline Formula/spirv-tools.rb:56]:
```suggestion
```
You never make use of this assignment, so there's no need to have it.

File: Formula/spirv-tools.rb [hunk 2]
```diff
@@ -40,7 +40,9 @@ def install
 
     mkdir "build" do
       system "cmake", "..", *std_cmake_args,
+                            "-DBUILD_SHARED_LIBS=ON",
                             "-DSPIRV_SKIP_TESTS=ON",
+                            "-DSPIRV_TOOLS_BUILD_STATIC=OFF",
```
Comment (carlocab) [inline Formula/spirv-tools.rb:45]: This looks like a pretty big change to the build. We should probably bump the revision here.

Comment (danielnachun) [reply Formula/spirv-tools.rb:45]: Good call, I will revision bump.

File: Formula/spirv-tools.rb (hunk 1 above)
Comment (danielnachun) [reply Formula/spirv-tools.rb:56]: `brew audit` fails with this error when I delete that line:

```
spirv-tools:
  * 58: col 7: Useless assignment to variable - `args`.
  * 62: col 7: Useless assignment to variable - `args`.
Error: 2 problems in 1 formula detected
```

File: Formula/spirv-tools.rb [hunk 3]
```diff
@@ -50,8 +52,19 @@ def install
 
   test do
     cp libexec/"examples"/"main.cpp", "test.cpp"
+
+    args = []
+
+    on_macos do
+      args = "-lc++"
+    end
```
Comment (carlocab) [inline Formula/spirv-tools.rb:60]:
```suggestion
    args = ["-lc++"]
```
To fix the audit, try this, along with removing the `args = []` above.

Comment (danielnachun) [reply Formula/spirv-tools.rb:60]: Ah that's the trick!  Thank you!

**Review by iMichka — APPROVED** [review]

**Pull request closed without merging**